packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -194,10 +203,12 @@ export class Dropdown implements ComponentInterface, DropdownInterface {
     return this.hostElement.shadowRoot.querySelector('slot');
   }
 
-  private disposeListener?: Function;
+  private disposeClickListener?: Function;
+  private disposeKeyListener?: Function;
```

Comment:
Avoid the `Function` type, as it provides little safety for the following reasons:
It provides no type safety when calling the value, which means it's easy to provide the wrong arguments.
It accepts class declarations, which will fail when called, as they are called without the new keyword.
([Source](https://typescript-eslint.io/rules/ban-types/))

e.g.

```
disposeClickListener?: () => void;
```